### PR TITLE
Update to LibHac 0.8.2

### PR DESCRIPTION
**Pull request by Thealexbarney**
This brings support for temporary savedata, ignores case in key names when loading from a file, and prints the rights ID correctly when missing a title key.

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/Fs/IFileSystemProxy.cs [hunk 1]
```diff
@@ -130,7 +130,7 @@ public ResultCode DeleteSaveDataFileSystem(ServiceCtx context)
         public ResultCode CreateSaveDataFileSystem(ServiceCtx context)
         {
             SaveDataAttribute  attribute      = context.RequestData.ReadStruct<SaveDataAttribute>();
-            SaveDataCreateInfo createInfo     = context.RequestData.ReadStruct<SaveDataCreateInfo>();
+            SaveDataCreationInfo creationInfo     = context.RequestData.ReadStruct<SaveDataCreationInfo>();
```
Comment (emmauss) [inline Ryujinx.HLE/HOS/Services/Fs/IFileSystemProxy.cs]: unnecessary whitespace

File: Ryujinx.HLE/HOS/Services/Fs/IFileSystemProxy.cs [hunk 2]
```diff
@@ -156,9 +156,9 @@ public ResultCode CreateSaveDataFileSystem(ServiceCtx context)
         public ResultCode CreateSaveDataFileSystemBySystemSaveDataId(ServiceCtx context)
         {
             SaveDataAttribute  attribute  = context.RequestData.ReadStruct<SaveDataAttribute>();
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/Fs/IFileSystemProxy.cs]: If you've decided to don't align vars as we did before, please remove extra spaces
